pandas/tests/util/test_assert_series_equal.py

```diff
@@ -36,13 +36,9 @@ def _assert_not_series_equal(a, b, **kwargs):
     kwargs : dict
         The arguments passed to `tm.assert_series_equal`.
     """
-    try:
+    msg = "The two Series were equal when they shouldn't have been"
+    with pytest.raises(AssertionError, match=msg):
         tm.assert_series_equal(a, b, **kwargs)
-        msg = "The two Series were equal when they shouldn't have been"
-
-        pytest.fail(msg=msg)
```

Comment:
I don't think removing this is entirely equivalent; might be root of CI failures